src/styles/index.scss

```diff
@@ -1,3 +1,63 @@
+@import '../utils/variables';
+
+* {
+  box-sizing: border-box;
+}
+
 body {
+  display: flex;
+  justify-content: center;
+  align-items: center;
+  height: 100vh;
   margin: 0;
+  padding: 0;
+}
+
+.calendar {
+  display: flex;
+  flex-wrap: wrap;
+  gap: $gap-columns;
+  max-width: ($day-size + $gap-columns) * $columns;
+
+  &__day {
+    display: flex;
+    justify-content: center;
+    align-items: center;
+    background-color: #eee;
+    height: $day-size;
+    width: $day-size;
+    border: 1px solid black;
+
+    @for $number from 1 through 31 {
+      &:nth-child(#{$number})::before {
+        content: '#{$number}';
+        font-family: Arial, Helvetica, sans-serif;
+        font-size: 30px;
+      }
+    }
+
+    &:hover {
+      cursor: pointer;
+      background-color: $bg-hover;
+      transform: translateY(-20px);
+      animation-name: transform;
+      animation-duration: 500ms;
+    }
+  }
+
+  @for $length from 28 through 31 {
+    &--month-length-#{$length - 1} {
+      .calendar__day:nth-child(n + #{$length}) {
+        display: none;
+      }
+    }
+  }
```

Comment:
The '&--month-length-#{$length - 1}' selector seems to be off by one. The naming suggests that it's for the previous month length, but the logic inside hides the days starting from the current length. Please verify if the naming aligns with the intended functionality.